### PR TITLE
Signup: limit site topic suggestions to 5

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

* add a `limit` prop to `SuggestionSearch`
* use a `limit` of 5 in the `site-topic` step to address too many results from dropping the preview way down the page as shown in this comment: https://github.com/Automattic/wp-calypso/pull/29125#issuecomment-444567029

#### Testing instructions

In the `site-topic` step in `/start/onboarding`, type something that would formerly provide > 5 suggestions. I suggest `me` as we found when starting to type "Mexican restaurant" in #28886.

The only other instance of `SuggestionSearch` is in the site topic picker in the `about` step of the `main` flow. Verify that its behaviour is unchanged (no limits)

### CODE REVIEW

File: client/components/suggestion-search/index.jsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ class SuggestionSearch extends Component {
 		onChange: PropTypes.func,
 		suggestions: PropTypes.array,
 		value: PropTypes.string,
+		limit: PropTypes.oneOfType( PropTypes.bool, PropTypes.number ),
```
Comment (ramonjd) [inline client/components/suggestion-search/index.jsx]: Do we need to accommodate two prop types? If the prop is passed down (and its value is a number > 0) it should be enough (?)

**Comment by scruffian** [comment]
LGTM 👍

File: client/components/suggestion-search/index.jsx (hunk 1 above)
Comment (southp) [reply client/components/suggestion-search/index.jsx]: +1. We could either go `limit = 0` as unlimited, or leave it as an optional prop :)

Comment (mattwiebe) [reply client/components/suggestion-search/index.jsx]: Good suggestion. I have a WP-wired brain that loves multiple typing in args and returns :)

**Comment by mattwiebe** [comment]
I added an enhancement in 644854d to put the search results that match the beginning of a typed string at the top of the list, followed by the rest which match in the middle of the word. I think this will improve the UX in all cases and particularly in those where the results are limited.

Eg, when typing `me` in the `onboarding/site-topic` step with the `limit` of 5 as in this PR, you will see:

image

…instead of the much-less helpful mid-word matches that come up: 

image

**Comment by michaeldcain** [comment]
> I added an enhancement in 644854d to put the search results that match the beginning of a typed string at the top of the list, followed by the rest which match in the middle of the word. I think this will improve the UX in all cases and particularly in those where the results are limited.

/cc @niranjan-uma-shanka to make sure that your endpoint work does something similar.

**Comment by mattwiebe** [comment]
Going to close this one in favour of #29201

**Pull request closed without merging**